docs/search/bidirectional.md

```diff
@@ -1,44 +1,48 @@
 author: FFjet, ChungZH, frank-xjh, hsfzLZH1, Xarfa, AndrewWayne
 
-### 双向同时搜索
+### Bidirectional simultaneous search
 
-从状态图上起点和终点同时开始进行宽度/深度优先搜索，如果发现相遇了，那么可以认为是获得了可行解。
+Start BFS/DFS from the starting node and ending node on the state graph at the same time. If the search meets, it can be assumed that a solution is obtained.
 
-双向广搜的步骤：
+The steps of bidirectional search:
 
 ```text
-开始结点 和 目标结点 入队列 q
-标记开始结点为 1
-标记目标结点为 2
-while(队列q不为空)
+starting_node and target_node enters queue q
+starting_node is marked as 1
+target_node is marked as 2
+while(queue q is not empty)
 {
-  从 q.front() 扩展出新的s个结点
+  expand s new nodes from q.front()
   
-  如果 新扩展出的结点已经被其他数字标记过
-    那么 表示搜索的两端碰撞
-    那么 循环结束
+  if   the newly expanded node has been marked by other numbers
+    then   both ends of the search meet
+    then   the loop is ended
   
-  如果 新的s个结点是从开始结点扩展来的
-    那么 将这个s个结点标记为1 并且入队q 
+  if   s new nodes are expanded from the starting node
+    then   mark s nodes as 1 and enter the queue q
     
-  如果 新的s个结点是从目标结点扩展来的
-    那么 将这个s个结点标记为2 并且入队q
+  if   s new nodes are expanded from the ending node
+    then   mark s nodes as 2 and enter the queue q
 }
 ```
 
-### 折半搜索
+### Meet in the Middle
 
-也称 meet in the middle，主要思想是将整个搜索过程分成两半，分别搜索，最后将两半的结果合并。由于搜索的复杂度往往是指数级的，而折半搜索可以使指数减半，也就能使复杂度开方。
+Meet in the middle is a searching technique that can be used when the input data size is small but is not small enough to use brute force solution. Its main idea is to divide the entire search process into two halves, search separately, and finally merge the results of the two halves. Since the time complexity of search is often exponential, a meet in the middle search can halve the exponent, and can also get the square root value the complexity.
 
-???+note "例题 [「USACO09NOV」灯 Lights](https://www.luogu.com.cn/problem/P2962)"
+???+note "Sample problem [「USACO09NOV」Lights](https://www.luogu.com.cn/problem/P2962) (original link in Chinese)"
 
-    有 $n$ 盏灯，每盏灯与若干盏灯相连，每盏灯上都有一个开关，如果按下一盏灯上的开关，这盏灯以及与之相连的所有灯的开关状态都会改变。一开始所有灯都是关着的，你需要将所有灯打开，求最小的按开关次数。
+    There are $n$ lamps. Each lamp is connected to several lamps, and each lamp has a switch. If you press the switch on one lamp, the switch status of this lamp and all the connected lamps will change. In the beginning, all the lights are off, and you need to turn on all the lights. Please find the minimum number of times to press the switch.
```

Comment:
```suggestion
    There are $n$ lamps. Each lamp is connected to several lamps, and each lamp has a switch. If you press the switch on a lamp, the switch status of this lamp and all lamps connected to it will change. In the beginning, all the lights are off, and you need to turn on all the lights. Please find the minimum number of operations.
```